### ISSUE TITLE
non-exhaustive patterns/ does not compile 

### ISSUE THREAD

**Issue by MariaGolubev** [standard-ai/tracing-gstreamer#6]
## Сompile time error :
```sh
➜  new-rtsp-service git:(main) ✗ cargo run
   Compiling tracing-gstreamer v0.3.0
   Compiling gstreamer-rtsp-server v0.18.0
error[E0004]: non-exhaustive patterns: `(&Kind(0_u8), _)` and `(&Kind(3_u8..=u8::MAX), _)` not covered
   --> /home/***/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-gstreamer-0.3.0/src/callsite.rs:152:38
    |
152 |             .then_with(move || match (&self.kind, &other.kind) {
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^ patterns `(&Kind(0_u8), _)` and `(&Kind(3_u8..=u8::MAX), _)` not covered
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
    = note: the matched value is of type `(&Kind, &Kind)`
```
### Cargo.toml:
```toml
[dependencies]
tracing = "0.1"
tracing-subscriber = "0.3"
tracing-gstreamer = "0.3"
log-panics = "2"


gstreamer-rtsp-server = "0.18"

url = "2"

tokio = {version = "1", features = ["rt-multi-thread", "macros", "sync"]}
```

**Comment by nagisa** [comment]
Could you share the version of the Rust toolchain (`rustc --version`) that was used here?

**Comment by MariaGolubev** [comment]
@nagisa 
```sh
➜  Stilsoft rustc --version
rustc 1.58.1 (db9d1b20b 2022-01-20)
```

**Comment by MariaGolubev** [comment]
@nagisa 
```sh
➜  Stilsoft rustup show        
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/golubev_mt/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
nightly-x86_64-unknown-linux-gnu

installed targets for active toolchain
--------------------------------------

wasm32-wasi
x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.58.1 (db9d1b20b 2022-01-20)
```

**Comment by nagisa** [comment]
cc https://github.com/tokio-rs/tracing/pull/1891

**Comment by nagisa** [comment]
Should be good to go now. The 0.2.2 and 0.3.1 releases cap the `tracing-core` version to the highest supported version and I'll release 0.3.2 shortly that includes the fix from #7 shortly.

I have yanked all the older releases that no longer work.
